cypress/e2e/crypto/crypto.spec.ts

```diff
@@ -134,169 +140,248 @@ const verify = function (this: CryptoTestContext) {
                             rel_type: "m.replace",
                             event_id: testEvent.event_id,
                         },
                     });
                 });
+
+                cy.get(".mx_EventTile_last .mx_EventTile_body")
+                    .within(() => {
+                        cy.findByText("Hee!");
+                    })
+                    .closest(".mx_EventTile")
+                    .should("have.class", "mx_EventTile_verified")
+                    .should("not.have.descendants", ".mx_EventTile_e2eIcon_warning");
             });
+        });
+    });
 
-            // the edit should have a warning
-            cy.contains(".mx_EventTile_body", "Haa!")
-                .closest(".mx_EventTile")
-                .within(() => {
-                    cy.get(".mx_EventTile_e2eIcon_warning").should("exist");
+    describe("Verify own device", () => {
+        let aliceBotClient: CypressBot;
+
+        beforeEach(function () {
+            cy.startHomeserver("default")
+                .as("homeserver")
+                .then((homeserver: HomeserverInstance) => {
+                    // Populate `win.matrixcs`
+                    cy.intercept("GET", "**/r0/login").as("login");
+                    cy.visit("/#/login");
+                    cy.wait("@login");
+
+                    // Create a new device for alice
+                    cy.getBot(homeserver, { bootstrapCrossSigning: true }).then((bot) => {
+                        aliceBotClient = bot;
+                    });
                 });
+        });
+
+        afterEach(function (this: CryptoTestContext) {
+            cy.stopHomeserver(this.homeserver);
+        });
 
-            // a second edit from the verified device should be ok
-            cy.get<ISendEventResponse>("@testEvent").then((testEvent) => {
-                this.bob.sendMessage(roomId, {
-                    "m.new_content": {
-                        msgtype: "m.text",
-                        body: "Hee!",
-                    },
-                    "m.relates_to": {
-                        rel_type: "m.replace",
-                        event_id: testEvent.event_id,
-                    },
+        function loginAlice(homeserver: HomeserverInstance) {
+            // Change homeserver
+            cy.findByRole("button", { name: "Edit" }).click();
+            // Select homeserver modal
+            cy.get(`[aria-labelledby="mx_BaseDialog_title"`).within(() => {
+                cy.findByRole("textbox").type(homeserver.baseUrl);
+                cy.findByRole("button", { name: "Continue" }).click();
+            });
+            // Fill credentials
+            cy.findByRole("textbox", { name: "Username" }).type(aliceBotClient.getUserId());
+            cy.findByLabelText("Password").type(aliceBotClient.__cypress_password);
+            // Sign in
+            cy.findByRole("button", { name: "Sign in" }).click();
+        }
+
+        function doAliceVerificationRequest() {
```

Comment:
this doesn't complete the verification request, so let's give it a better name. And a comment!


```suggestion
        /* Click the "Verify with another device" button, and have the bot client auto-accept it.
         *
         * Stores the incoming `VerificationRequest` on the bot client as `@verificationRequest`.
         */
        function initiateAliceVerificationRequest(): void {
```

Reply:
03a45fc8ea